be/src/fs/fs_s3.cpp

```diff
@@ -56,7 +56,20 @@ bool operator==(const Aws::Client::ClientConfiguration& lhs, const Aws::Client::
 
 class S3ClientFactory {
 public:
-    using ClientConfiguration = Aws::Client::ClientConfiguration;
+    // add bucket and credential infos to ClientConfiguration can prevent invalid _clients[i] use
+    // when bueket or credential change but properties of Aws::Client::ClientConfiguration unchanged
+    struct ClientConfiguration {
+        Aws::Client::ClientConfiguration client_config;
+        std::string bucket;
+        std::string access_key_id;
+        std::string secret_access_key;
+
+        bool operator==(const ClientConfiguration& rhs) {
+            return client_config == rhs.client_config && bucket == rhs.bucket && access_key_id == rhs.access_key_id &&
```

Comment:
It's very wierd to have `bucket` field in ClientConfiguration. And access_key_id and access_secret_key are already in `AWS::Client::ClientConfiguration`.

Reply:
`AWS::Client::ClientConfiguration` contains only SSL related info but no ak/sk.

bucket is removed from ClientConfiguration.